include/micm/system/phase.hpp

```diff
@@ -10,18 +10,35 @@
 namespace micm
 {
 
+  /**
+   * @brief A class which represnts a certain chemical phase (gaseous, aquous)
+   * 
+   * Each phase represents a set of species that participate in chemical reactions in that phase.
+   */
+  template<typename T>
   class Phase
   {
    private:
-    const std::vector<Species> species_;
+    /// @brief The list of species
+    const std::vector<Species<T>> species_;
 
    public:
-    Phase();
+    /// @brief Default constructor
+    Phase() = default;
+    /// @brief Phase copy constructor
+    /// @param other Another micm::Phase
     Phase(const Phase& other);
-    Phase(std::vector<Species> species);
+    /// @brief Create a phase with a set of species
+    /// @param species A unique list of species
+    Phase(std::vector<Species<T>> species);
 
+    /// @brief Assignment operator. This assumes ownership of the other's list of species
+    /// @param other Another micm::Phase
+    /// @return a reference to the created phase
     Phase& operator=(const Phase& other);
```

Comment:
should these be defined here now that the `cpp` file has been removed?

Reply:
I hadn't because I figured defining these would have been writing code and I was planning on leaving that for later. They will have very simple implementations so I can define them now and fully write the tests for them if you'd like me to